### PR TITLE
fix `ReifiedGeneric` for pyright

### PR DESCRIPTION
**Pull request by DetachHead**
changenotes:
- BREAKING: `T_cont` renamed to `T_contra`
- BREAKING: drop support for 3.7

### CODE REVIEW

File: basedtyping/__init__.py [hunk 1]
```diff
@@ -240,22 +241,27 @@ def __call__(cls, *args: object, **kwargs: object) -> object:
...
 GenericItems: TypeAlias = Union[type, TypeVar, Tuple[Union[type, TypeVar], ...]]
 """The ``items`` argument passed to ``__class_getitem__`` when creating or using a ``Generic``"""
 
 
-class ReifiedGeneric(Generic[T], metaclass=_ReifiedGenericMetaclass):
+class ReifiedGeneric(
+    Generic[T],
+    # mypy doesn't support metaclasses with generics but for pyright we need to correctly type the `__call__`
+    # return type, otherwise all instances of `ReifiedGeneric` will have the wrong type
+    metaclass=_ReifiedGenericMetaclass[Self],  # type:ignore[misc]
```
Comment (KotlinIsland) [inline basedtyping/__init__.py:255]: works?

File: pyproject.toml [hunk 2]
```diff
@@ -8,10 +8,10 @@ name = "basedtyping"
 version = "0.0.3"
 
 [tool.poetry.dependencies]
-python = "^3.7.2"
+python = "^3.8"
```
Comment (KotlinIsland) [inline pyproject.toml:11]: Can we update the ci matrix to drop 3.7

**Review by KotlinIsland — APPROVED** [review]

File: basedtyping/__init__.py (hunk 1 above)
Comment (DetachHead) [reply basedtyping/__init__.py:255]: yep. added a test

File: .github/workflows/check.yaml [hunk 3]
```diff
@@ -10,7 +10,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11-dev"]
+        python-version: ["3.8", "3.9", "3.10", "3.11-dev"]
```
Comment (KotlinIsland) [inline .github/workflows/check.yaml:13]:
```suggestion
        python-version: ["3.8", "3.9", "3.10", "3.11"]
```

**Review by KotlinIsland — APPROVED** [review]

**Pull request merged**